app/controller.js

```diff
@@ -306,6 +307,26 @@ export default function(state, emitter) {
     render();
   });
 
+  emitter.on('report', async ({ reason }) => {
+    try {
+      const file = state.fileInfo;
+      if (!file) {
+        // TODO
+        emitter.emit('pushState', '/error');
+        return render();
+      }
+      await reportLink(file.id, file.secretKey, reason);
+      render();
+    } catch (err) {
+      console.error(err);
+      if (err.message === '404') {
```

Comment:
Does this actually happen every time?

Reply:
No, 

`/api/report` returns

* 200 - when the link was already flagged or was successfully flagged. Note that the file may already be "dead"
* 400 - when the submitted `key` can't decrypt the metadata. This is to prevent reporting ids without knowing the full url
* 404 - when the id has fully expired and deleted from redis. An api delete just marks the id as dead now so that we can accept reports for files that have already hit max download count

If it's a 404 we don't want to say there was an error